BetaCameras/RealSense2/RealSense2.cs

```diff
@@ -1846,9 +1859,31 @@ unsafe public override IProjectiveTransformation GetIntrinsics(string channelNam
                 intrinsics.coeffs[2],
                 intrinsics.coeffs[3]);
 
+            intrinsicsCache[keyName] = projTrans;
+
             return projTrans;
         }
 
+        private Point2i GetResolutionFromChannelName(string channelName)
+        {
+            Point2i resolution = new Point2i(640, 480);
+            switch (channelName)
+            {
+                case ChannelNames.Color:
+                    resolution = ColorResolution;
+                    break;
+
+                case ChannelNames.Left:
+                case ChannelNames.Right:
+                case ChannelNames.Distance:
+                case ChannelNames.ZImage:
+                default:
+                    resolution = DepthResolution;
+                    break;
+            }
+            return resolution;
```

Comment:
`return` in the `default` case, and remove the last line (then unreachable code)